pysilcam/silcamgui/interactive_summary.py

```diff
@@ -461,76 +500,88 @@ def toggle_plot(self):
+            self.axistext.clear()
+            self.axistext.text(1, 1, string, va='top', ha='right', transform=self.axistext.transAxes)
+            self.axistext.axis('off')
 
-            plt.sca(self.axisconstant)
+            #plt.sca(self.axisconstant)
             self.line1.remove()
             self.line2.remove()
-            self.line1 = plt.vlines(start_time, self.yrange[0], self.yrange[1], 'r', linestyle='--')
-            self.line2 = plt.vlines(end_time, self.yrange[0], self.yrange[1], 'r', linestyle='--')
+            self.line1 = self.axisconstant.vlines(start_time, self.yrange[0], self.yrange[1], 'r', linestyle='--')
+            self.line2 = self.axisconstant.vlines(end_time, self.yrange[0], self.yrange[1], 'r', linestyle='--')
             self.canvas.draw()
             return
 
-        psd_start = min(u[timeind])
-        psd_end = max(u[timeind])
+        ds_psd = ds.sel(time=slice(start_time, end_time)).mean(dim='time')
 
-        psd_total = np.mean(self.vd_total[timeind, :], axis=0)[0]
-        psd_oil = np.mean(self.vd_oil[timeind, :], axis=0)[0]
-        psd_gas = np.mean(self.vd_gas[timeind, :], axis=0)[0]
+        psd_start = min(ds_slice['time'].values)
+        psd_end = max(ds_slice['time'].values)
 
-        psd_vc_total = np.sum(psd_total)
-        psd_vc_oil = np.sum(psd_oil)
-        psd_vc_gas = np.sum(psd_gas)
+        psd_vc_total = np.sum(ds_psd['vd_total'].values)
+        psd_vc_oil = np.sum(ds_psd['vd_oil'].values)
+        psd_vc_gas = np.sum(ds_psd['vd_gas'].values)
 
-        psd_peak_total = self.dias[np.argwhere(psd_total == max(psd_total))][0][0]
-        psd_peak_oil = self.dias[np.argwhere(psd_oil == max(psd_oil))][0][0]
-        psd_peak_gas = self.dias[np.argwhere(psd_gas == max(psd_gas))][0][0]
+        # Use xarray idxmax function to find coordinate label (e.g. size bin) of maximum PSD value
+        psd_peak = ds_psd.idxmax()
+        psd_peak_total = psd_peak['vd_total'].values
+        psd_peak_oil = psd_peak['vd_oil'].values
+        psd_peak_gas = psd_peak['vd_gas'].values
 
-        psd_d50_total = scpp.d50_from_vd(psd_total, self.dias)
-        psd_d50_oil = scpp.d50_from_vd(psd_oil, self.dias)
-        psd_d50_gas = scpp.d50_from_vd(psd_gas, self.dias)
+        psd_d50_total = scpp.d50_from_vd(ds_psd['vd_total'], self.dias)
+        psd_d50_oil = scpp.d50_from_vd(ds_psd['vd_oil'], self.dias)
+        psd_d50_gas = scpp.d50_from_vd(ds_psd['vd_gas'], self.dias)
 
-        psd_gor = sum(psd_gas) / (sum(psd_oil) + sum(psd_gas)) * 100
+        psd_gor = sum(ds_psd['vd_gas'].values) / (sum(ds_psd['vd_oil'].values) + sum(ds_psd['vd_gas'].values)) * 100
 
-        plt.sca(self.axispsd)
-        plt.cla()
-        plt.plot(self.dias, psd_total, 'k', linewidth=5, label='Total')
-        plt.plot(self.dias, psd_oil, color=[0.7, 0.4, 0], label='Oil')
-        plt.plot(self.dias, psd_gas, 'b', label='Gas')
+        #plt.sca(self.axispsd)
```

Comment:
E265: block comment should start with '# '  <!-- Automatically posted by Lintly -->